src/ci.js

```diff
@@ -76,9 +76,9 @@ exports.getVars = function() {
   // Semaphore
   if (env.CI === 'true' && env.SEMAPHORE === 'true') {
     return {
-      build: env.SEMAPHORE_BUILD_NUMBER,
-      branch: env.BRANCH_NAME,
-      commit: env.REVISION
+      build: env.SEMAPHORE_BUILD_NUMBER || env.SEMAPHORE_WORKFLOW_ID,
+      branch: env.BRANCH_NAME || env.SEMAPHORE_GIT_BRANCH,
+      commit: env.REVISION || env.SEMAPHORE_GIT_PR_SHA
```

Comment:
After implementing a workaround in our repo, I wonder if this should be `SEMAPHORE_GIT_SHA` ? 

When I tried `SEMAPHORE_GIT_PR_SHA` Screener would fail to run.

Reply:
This worked for me when I tested, but TBH the documentation is a little... unclear.

On SEMAPHORE_GIT_SHA, the docs say 

> The value of the SEMAPHORE_GIT_SHA environment variable holds the current revision of code that the pipeline is using.

For SEMAPHORE_GIT_PR_SHA, the docs say

> The value of the SEMAPHORE_GIT_PR_SHA environment variable holds the commit SHA of the HEAD commit of the Pull Request.

I figured that you're comparing a PR to Master so that you know if it's safe to merge, so the PR SHA is the desirable value here (So that Screener can update the PR's status widget)... But it's weird this didn't run for you.  Did Screener kick off at all (eg, in the Screener interface) and just not update the widget, or did it not run at all, @mikrotron ?

Reply:
@DylanLacey `SEMAPHORE_GIT_PR_SHA` will only be available when it's a Pull Request, otherwise it's undefined. Can you update to `SEMAPHORE_GIT_SHA`? Or cover all cases:

`env.REVISION || env.SEMAPHORE_GIT_PR_SHA || env.SEMAPHORE_GIT_SHA`

Also, as @mikrotron linked above, this is already working in their Semaphore CI builds.

Reply:
> `SEMAPHORE_GIT_PR_SHA` will only be available when it's a Pull Request, otherwise it's undefined. 

That makes sense. I was pushing commits _without_ an open PR to test if it worked, and that's when I noticed it wasn't integrating with GitHub.  The Screener build itself worked fine, it just never showed up in GitHub. 

This is the workaround that we used to get this to work for us in the meantime:
https://github.com/acl-services/paprika/blob/master/.semaphore/semaphore.yml#L98